components/job-orchestration/job_orchestration/scheduler/job_config.py

```
@@ -44,6 +45,22 @@ def validate_keys(cls, value):
+    type: Literal[InputType.METADATA.value] = InputType.METADATA.value
+    ingestion_job_id: int
+    dataset: str | None = None
+    timestamp_key: str | None = None
+    unstructured: bool = False
+    metadata_ids: list[int]
+
+    @field_validator("metadata_ids")
+    @classmethod
+    def validate_metadata_ids_non_empty(cls, value: list[int]) -> list[int]:
+        if len(value) == 0:
+            raise ValueError("metadata_ids cannot be an empty list")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)